content/automate/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/start_retirement.rb

```diff
@@ -17,7 +17,7 @@
   exit MIQ_ABORT
 end
 
-if vm.retiring?
```

Comment:
@d-m-u We need 3 separate checks in all of the start_retirement methods.
The methods already have 2 of the checks 
 
```
if stack.retired?
  $evm.log('error', "Stack is already retired. Aborting current State Machine.")
  exit MIQ_ABORT
end

if stack.retiring?
  $evm.log('error', "Stack is in the process of being retired. Aborting current State Machine.")
  exit MIQ_ABORT
end
```

We need to add this check:
 
```
unless unless stack.retirement_initialized?
  $evm.log('error', "Stack has not been initialized for retirement. Aborting current State Machine.")
  exit MIQ_ABORT
end
```